source/custom/visa_service.custom.cpp

```diff
@@ -320,28 +320,25 @@ ::grpc::Status VisaService::Read(::grpc::ServerContext* context, const ReadReque
   if (context->IsCancelled()) {
     return ::grpc::Status::CANCELLED;
   }
-#if 1
-  return ::grpc::Status(grpc::StatusCode::DO_NOT_USE, "Custom code not implemented yet");
-#else
   try {
     auto vi_grpc_session = request->vi();
     ViSession vi = session_repository_->access_session(vi_grpc_session.name());
     ViUInt32 count = request->count();
-    std::string buffer(return_count, '\0');
+    std::string buffer(256 - 1, '\0');
     ViUInt32 return_count{};
     auto status = library_->Read(vi, (ViByte*)buffer.data(), count, &return_count);
-    if (!status_ok(status)) {
+    if (!status_ok(status) && return_count == 0) {
       return ConvertApiErrorStatusForViSession(context, status, vi);
     }
     response->set_status(status);
+    buffer.resize(return_count);
     response->set_buffer(buffer);
```

Comment:
Don't bother resizing the buffer. Instead, use the explicit:
`response->set-buffer(buffer.data(), return_count);`